pulsar-broker/src/main/java/org/apache/pulsar/broker/systopic/LedgerDeletionSystemTopicClient.java

```diff
@@ -0,0 +1,220 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.systopic;
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import org.apache.bookkeeper.mledger.deletion.PendingDeleteLedgerInfo;
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.DeadLetterPolicy;
+import org.apache.pulsar.client.api.Message;
+import org.apache.pulsar.client.api.MessageId;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.SubscriptionType;
+import org.apache.pulsar.client.api.TypedMessageBuilder;
+import org.apache.pulsar.client.impl.ConsumerImpl;
+import org.apache.pulsar.client.impl.MultiTopicsConsumerImpl;
+import org.apache.pulsar.common.naming.SystemTopicNames;
+import org.apache.pulsar.common.naming.TopicName;
+import org.apache.pulsar.common.util.FutureUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * System topic for ledger deletion.
+ */
+public class LedgerDeletionSystemTopicClient extends SystemTopicClientBase<PendingDeleteLedgerInfo> {
+
+    private final int sendDelaySeconds;
+
+    private final int reconsumeLaterSeconds;
+
+    public LedgerDeletionSystemTopicClient(PulsarClient client, TopicName topicName, int sendDelaySeconds,
+                                           int reconsumeLaterSeconds) {
+        super(client, topicName);
+        this.sendDelaySeconds = sendDelaySeconds;
+        this.reconsumeLaterSeconds = reconsumeLaterSeconds;
+    }
+
+    @Override
+    protected CompletableFuture<Writer<PendingDeleteLedgerInfo>> newWriterAsyncInternal() {
+        return client.newProducer(Schema.AVRO(PendingDeleteLedgerInfo.class))
+                .topic(topicName.toString())
+                .enableBatching(false)
+                .createAsync().thenCompose(producer -> {
+                    if (log.isDebugEnabled()) {
+                        log.debug("[{}] A new writer is created", topicName);
+                    }
+                    return CompletableFuture.completedFuture(
+                            new PendingDeleteLedgerWriter(producer, LedgerDeletionSystemTopicClient.this,
+                                    sendDelaySeconds));
+                });
+    }
+
+    @Override
+    protected CompletableFuture<Reader<PendingDeleteLedgerInfo>> newReaderAsyncInternal() {
+        // TODO: 2022/7/7 死信队列发消息batch情况
+        return client.newConsumer(Schema.AVRO(PendingDeleteLedgerInfo.class))
```

Comment:
Here, you should enable retry, If you want to use `reconsumeLater` .

Reply:
thanks